bootstrap/bootstrap-storage.sh

```diff
@@ -41,60 +40,57 @@ else
 fi
 
-TABLE_INITIALIZER_MAIN_CLASS=com.hortonworks.streamline.storage.tool.TablesInitializer
-for file in "${BOOTSTRAP_DIR}"/lib/*.jar;
+if grep -E "^(\s)*db.type:(\s)*\"mysql\"(\s)*$" $CONFIG_FILE_PATH > /dev/null; then
+  if [ ! -f ../libs/$MYSQL_JAR_PATTERN ] ; then
+    echo "MySQL java connector jar not found in libs, downloading the jar from $MYSQL_JAR_URL_PATH .."
+    TEMP_DIR=`mktemp -d`
+    MYSQL_CONNECTOR_DIR="mysql-connector-java"
+    ZIP_FILE_NAME="$TEMP_DIR/$MYSQL_CONNECTOR_DIR.zip"
+    curl -o $ZIP_FILE_NAME -LOk $MYSQL_JAR_URL_PATH > /dev/null 2>&1
+    unzip $ZIP_FILE_NAME -d $TEMP_DIR/$MYSQL_CONNECTOR_DIR > /dev/null
+    cp $TEMP_DIR/$MYSQL_CONNECTOR_DIR/*/$MYSQL_JAR_PATTERN ../libs
+    rm -rf $TEMP_DIR
+    echo "Successfully copied MySQL connector jar to libs"
+  fi
+fi
+
+for file in "${BOOTSTRAP_DIR}"/../libs/*.jar;
 do
     CLASSPATH="$CLASSPATH":"$file"
 done
 
-echo "Using Configuration file: ${CONFIG_FILE_PATH}"
-
-function dropTables {
-    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --drop
-}
-
-function createTables {
-    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --create
-}
-
-function checkStorageConnection {
-    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --check-connection
-}
-
 function printUsage {
-    echo "USAGE: $0 [create|drop|check-connection|drop-create]"
+    cat <<-EOF
+USAGE: $0 [baseline|migrate|info|validate|clean]
+   baseline : Use this option, if the target database doesn't have DATABASE_CHANGE_LOG table and it already has some migrations.
+              Ex:- Lets say we have a total of 3 migrations with the latest version of schema registry, the target database already has version 1 and 2 and we only need to apply migration 3.
+              We need to create a new DATABASE_CHANGE_LOG and mark migration 1 and 2 as applied, for that add a parameter "baselineVersion" under "db.migration" in streamline.yaml and set it to 2,
+              i.e that latest version of the target database. Invoke bootstrap-storage.sh script with baseline option. This will create DATABASE_CHANGE_LOG table and mark migration 2 and below are applied.
+              After this invoke bootstrap-storage.sh with migration option, this run migration 3 while ignoring migration 2 and 1.
+   migrate  : Runs the migration on the target database from the last checkpoint, if DATABASE_CHANGE_LOG is absent then creates the DATABASE_CHANGE_LOG table and runs all the migrations
+   info     : Shows the status of all the migrations
+   validate : Validate if the all the migrations are applied on the target database
+   clean    : Drop all the tables in the target database
```

Comment:
IMO, we should retain all the previous options with bootstrap-storage since users would be already familiar with those options. We only need a new option "migrate" (and maybe info/validate).